Library/Homebrew/cask/artifact/abstract_uninstall.rb

```diff
@@ -73,7 +75,7 @@ def dispatch_uninstall_directive(directive_sym, **options)
 
         args = directives[directive_sym]
 
-        send("uninstall_#{directive_sym}", *(args.is_a?(Hash) ? [args] : args), **options)
+        send("uninstall_#{directive_sym}", (args.is_a?(Enumerable) ? args : [args]), **options)
```

Comment:
This previously converted args to a splat, in order to subsequently call a method that accepts splat arguments. I've cut out the middleman, in order to enable typing, although I think it also makes for slightly more understandable code.

The methods below are all private, which ofc doesn't guarantee that changing the method signature is safe, so this proposed change deserves a critical eye.

Reply:
Isn't `Hash` also an `Enumerable`, i.e. would now be just `args` instead of `[args]`?

Reply:
@reitermarkus Great question! When it was a Hash, it would be wrapped as an array, then splatted back into a hash arg. Since we've removed the `*`, we no longer want to wrap hashes as arrays.

This might help explain things:
```ruby
def foo(args, c: nil)
  puts args
end

argv = 'str'
foo(*argv)
# => No error
argv = {a: 1, b: 2}
foo(*argv)
# => ArgumentError: wrong number of arguments (given 2, expected 1)
foo(*[argv])
# => No error
```
Ruby will convert a hash into kwargs when preceded by `*`, leading to errors in methods that take a mix of positional and kwargs (iirc without the kwargs it will pass the hash as the first poisitional arg). You can try it for yourself by changing `Enumerable` to `Array` above, and running `brew tests --only=cask/artifact/uninstall`. 

(I believe this also relates to why sorbet has a hard time supporting splats, but my memory is hazy here)

I generally find this all pretty hard to wrap my head around, so I try to avoid splats regardless of Sorbet's stance 🤕

Reply:
Previously, we had e.g.

```ruby
def uninstall_something(*args)
  args
end
```

and passing a `Hash` would look like

```ruby
uninstall_something(*[{a: 1}]) == [{a: 1}]
````

while now we have

```ruby
def uninstall_something(args)
  args
end
```

and passing a `Hash`, since it is `Enumerable`, we effectively have

```ruby
uninstall_something({a: 1}) == {a: 1}
```

which isn't the same thing.

Reply:
Okay, so the only method we actually pass a `Hash` to is `uninstall_script`, which was not using a splat to begin with, so passing both `*[{a: 1}]` and `{a: 1}` still behave the same for that case.